### ISSUE TITLE
Create a usage scenario

### ISSUE THREAD

**Issue by QuentinDaveau** [med-material/Whack_A_Mole_VR#15]
It would be nice to write down a usage scenario of a basic use of the software, this would help to spot issues with the current design, and to specify imprecise mechanics. Here's my take on the scenario:

**Actors**: the participant, the therapist.

**Goal**: help a therapist to find blind spots in the field of view of the participant in an interactive and entertainig way.

**Usage scenario:**
- The participant puts the Vive. In the game, he is welcomed by a panel explainig him the basics on the game, and telling him to wait fot the therapist to configure the game. He cannot interact with anything.
- The therapist sets up the game. He defines the participant Id, the game duration (with default value), and the test Id. He also defines the difficulty of the game.
- The therapist starts the game. The welcoming panel disappears and the moles starts to activate.
- The participant pops the moles until the game time is over. There are fake moles which shouldn't be pop.
- Once the game is over, logged datas are sent to a server, and the therapist gets a visual feedback of the results.

This already raises quite a few issues with the current state of the game, it would be nice to discuss the scenario and the possible issues before raising git issues.

**Comment by bastianilso** [comment]
Thanks for writing this up, Daveau. I generally agree that there is some way to go until the software can be used by therapists and patients. Could you provide some examples here of the issues you have observed with the current behavior and suggestions?

I suggest we move the usage scenario(s) to the [README.md](https://github.com/med-material/Whack_A_Mole_VR/edit/master/README.md) file so it is stored persistently.

**Comment by QuentinDaveau** [comment]
Some issues I thought about:

- Currently the game stops after a certain number of missed moles. This gets in conflict with the set game duration and the goal of dathering datas. This feature should be removed and the game should simply stops once the time is over. Maybe it can be implemented in an "endless" mode.

- If the goal of the game is to spot blind spots in the participant's field of view, is would be necessary to assure that he is always looking at the same spot on the moles' wall (that he can't move the head to look around). Otherwise the datas could be biased. To do so we could either fix the wall position to the vive look direction (which would be very disturbing), or empathize the participant to look at a given spot in an indirect way (secondary goal ? May need design) .It may be possible to keep the game as-is and to focus on reaction time (time spent before an active mole gets pop), but it may be less precise. Either way the duration between the activation and popping of a given mole could be useful information.

- If there is a possibility to set a test number, it should be unique, at least compared to the participant's other tests. It could easily be done once the game is able to communicate with the database.

Some imprecisions raised by the scenario:

- We know that the therapist has an UI to control the game. We should define the main features of this UI. Right now it is defined that the therapist has two menu, one dedicated to the game setup (game duration, test Id, participant Id, difficulty, start game) and one always active for in-game modifications (mirror effect, prism effect, eye patch, dual task mode, stop test, time passed, time left). Should any other feature be added to the menu ?

**Comment by bastianilso** [comment]
>  Currently the game stops after a certain number of missed moles. 
>  This gets in conflict with the set game duration and the goal of dathering datas. 
> This feature should be removed and the game should simply stops once the time is over. Maybe it can be implemented in an "endless" mode.
> 
I agree. Please do this in a separate branch, as a separat git commit and submit as a separate pull request. Then we can rebase your work on top.

>  If the goal of the game is to spot blind spots in the participant's field of view, is would be necessary to assure that he is always looking at the same spot on the moles' wall (that he can't move the head to look around). Otherwise the datas could be biased. To do so we could either fix the wall position to the vive look direction (which would be very disturbing), or empathize the participant to look at a given spot in an indirect way (secondary goal ? May need design) .It may be possible to keep the game as-is and to focus on reaction time (time spent before an active mole gets pop), but it may be less precise. Either way the duration between the activation and popping of a given mole could be useful information.
> 
The patients' deficit is a bit more complicated than so. We let them look around on purpose. But, I agree with reaction time - It would be useful to perform data visualization in R Shiny of that. I think we can calculate that in R by knowing the timestamps between two TargetHit events.

> If there is a possibility to set a test number, it should be unique, at least compared to the participant's other tests. It could easily be done once the game is able to communicate with the database.
> 
The test number was a number which an experiment facilitator could set in case several tests are run on the same participant ID. That said, I think its a good idea to also have a unique identifier. Maybe we can generate an MD5 shorthand of participantID+testNumber+date? (In that case please commit this as part of the Logger: Log Basic Information pull request)


> Some imprecisions raised by the scenario:
> 
> We know that the therapist has an UI to control the game. We should define the main features of this UI. Right now it is defined that the therapist has two menu, one dedicated to the game setup (game duration, test Id, participant Id, difficulty, start game) and one always active for in-game modifications (mirror effect, prism effect, eye patch, dual task mode, stop test, time passed, time left). Should any other feature be added to the menu ?
> 
I think the therapist menu needs a design, and suggest we open a separate issue for that. Currently the sidebar menu does a decent job, but I would like to see if I can come up with something better.

**Comment by QuentinDaveau** [comment]
> I agree. Please do this in a separate branch, as a separat git commit and submit as a separate pull request. Then we can rebase your work on top.

Okay. I'll add it to the code rework since it is closely linked to the Game manager working.

I'll also add the unique identifier and reaction time to the issue #2.

I'll create a new issue for the therapist UI.
